code/modules/mob/death.dm

```diff
@@ -76,6 +76,11 @@
 	for(var/obj/item/implant/carrion_spider/control/C in src)
 		C.return_mind()
 
+	// Notifies borer of host death.
+	var/mob/living/simple_animal/borer/B = has_brain_worms()
+	if(B?.controlling)
+		B.host_death()
```

Comment:
`has_brain_worms` returns `FALSE` instead of `null` when there is no borer so it runtimes

```
[18:13:00] Runtime in death.dm,81: Cannot read 0.controlling
  proc name: death (/mob/proc/death)
  src: the space carp (/mob/living/simple_animal/hostile/carp)
  src.loc: the underplating (77,11,13) (/turf/simulated/floor/plating/under/airless)
  call stack:
  the space carp (/mob/living/simple_animal/hostile/carp): death(null, "dies!", "You have died.")
  the space carp (/mob/living/simple_animal/hostile/carp): death(null, "dies!")
  the junk tractor beam generato... (/obj/jtb_generator): generate edge()
  the junk tractor beam generato... (/obj/jtb_generator): populate z level()
  the junk tractor beam generato... (/obj/jtb_generator): generate junk field()
  Misc (/datum/controller/subsystem/misc): build junk field()
  Misc (/datum/controller/subsystem/misc): Initialize(null)
  Master (/datum/controller/master): Initialize(10, 0, null)
```